sale_stock_picking_blocking/models/sale_order.py

```diff
@@ -26,11 +26,17 @@ def _check_not_auto_done(self):
                 _('You cannot block a sale order with "auto_done_setting" ' "active.")
             )
 
-    @api.depends("partner_id")
+    @api.depends("partner_id", "payment_term_id")
     def _compute_delivery_block_id(self):
-        """Add the 'Default Delivery Block Reason' if set in the partner."""
+        """Add the 'Default Delivery Block Reason' if set in the partner
+        or in the payment term."""
         for so in self:
-            so.delivery_block_id = so.partner_id.default_delivery_block or False
+            if so.partner_id.default_delivery_block:
```

Comment:
I don't agree on this. As the change of value on partner will trigger a recompute of all sale orders. Which is unwanted.

Reply:
IMHO, it's the same for default_delivery_black_reason_id on payment_term_id

Reply:
@rousseldenis Nice warning, indeed now that you mention, it makes more sense to keep the value unchanged in the SO as it could have been manually modified even.

Reply:
@LoisRForgeFlow @rousseldenis This has now been corrected.